kratos/includes/kratos_parameters.h

```diff
@@ -712,6 +749,14 @@ class KRATOS_API(KRATOS_CORE) Parameters
      */
     void RecursivelyValidateAndAssignDefaults(Parameters& rDefaultParameters);
 
+    /**
+     * @brief This function is designed to verify that the parameters under testing match the form prescribed by the defaults. (const version)
+     * @details If the parameters contain values that do not appear in the defaults, an error is thrown, whereas if a parameter is found in the defaults but not in the Parameters been tested, it is copied to the parameters.
+     * This version walks and validates the entire json tree below the point at which the function is called
+     * @param rDefaultParameters Parameters of reference which we use to check
+     */
+    void RecursivelyValidateAndAssignDefaults(const Parameters& rDefaultParameters);
```

Comment:
I think this method should only accept a const object, why would it need to modify the input?

Reply:
I would not change the current behavior right now